spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala

```diff
@@ -0,0 +1,143 @@
+    log.info(s"Input schema: ${streamData.schema.prettyJson}")
+  }
+
+  @throws[IllegalArgumentException]
+  private def getReportVersion(implicit cmd: ConfCmdConfig): Int = {
+    cmd.reportVersion match {
+      case Some(version) => version
+      case None => throw new IllegalArgumentException("Report version is not provided.")
+    }
+  }
+}
+
+object HyperConformance extends StreamTransformerFactory {
+  val log: Logger = LoggerFactory.getLogger(this.getClass)
+
+  // Configuration keys expected to be set up when running Conformance as a Transformer component for Hyperdrive
+  val menasUriKey = "menas.rest.uri"
+  val menasCredentialsFileKey = "menas.credentials.file"
+  val menasAuthKeytabKey = "menas.auth.keytab"
+  val datasetNameKey = "dataset.name"
+  val datasetVersionKey = "dataset.version"
+  val reportDateKey = "report.date"
+  val reportVersionKey = "report.version"
+
+  @throws[IllegalArgumentException]
+  override def apply(conf: Configuration): StreamTransformer = {
+    log.info("Building HyperConformance")
+
+    validateConfiguration(conf)
+
+    val menasCredentialsFactory = getMenasCredentialsFactory(conf: Configuration)
+
+    implicit val cmd: ConfCmdConfig = ConfCmdConfig(
+      datasetName = conf.getString(datasetNameKey),
+      datasetVersion = conf.getInt(datasetVersionKey),
+      reportDate = conf.getString(reportDateKey),
+      reportVersion = Option(conf.getInt(reportVersionKey)),
+      menasCredentialsFactory = menasCredentialsFactory,
+      performanceMetricsFile = None,
+      publishPathOverride = None,
+      folderPrefix = None,
+      experimentalMappingRule = Some(true),
+      isCatalystWorkaroundEnabled = Some(true),
+      autocleanStandardizedFolder = Some(false),
+      persistStorageLevel = None
+    )
+
+    implicit val featureSwitcher: FeatureSwitches = FeatureSwitches()
+      .setExperimentalMappingRuleEnabled(true)
+      .setCatalystWorkaroundEnabled(true)
+      .setControlFrameworkEnabled(false)
+      .setBroadcastStrategyMode(Always)
+      .setBroadcastMaxSizeMb(0)
+
+    implicit val menasBaseUrls: List[String] = MenasConnectionStringParser.parse(conf.getString(menasUriKey))
+
+    new HyperConformance()
+  }
+
+  @throws[IllegalArgumentException]
+  def validateConfiguration(conf: Configuration): Unit = {
+    val mandatoryKeys = List(menasUriKey, datasetNameKey, datasetVersionKey, reportDateKey, reportVersionKey)
+
+    val missingKeys = mandatoryKeys.filterNot(key => conf.containsKey(key))
+
+    if (missingKeys.nonEmpty) {
+      throw new IllegalArgumentException(s"Missing mandatory configuration parameters for keys: ${missingKeys.mkString(", ")}.")
+    }
+  }
+
+  @throws[IllegalArgumentException]
+  private def getMenasCredentialsFactory(conf: Configuration): MenasCredentialsFactory = {
+    val hasCredentialsFile = conf.containsKey(menasCredentialsFileKey)
+    val hasKeytab = conf.containsKey(menasAuthKeytabKey)
+
+    (hasCredentialsFile, hasKeytab) match {
```

Comment:
Shouldn't the error cases be part of the validation (`validateConfiguration`)?

Reply:
Thanks for the suggestion. I still prefer it this way.

Reply:
Fair enough. Not a hard requirement from my side. :-) Just thought that having the exceptions from one place could be easier for handling, that's all.